mesonbuild/interpreter/interpreter.py

```diff
@@ -2801,6 +2801,12 @@ def extract_incdirs(self, kwargs, key: str = 'include_directories') -> T.List[bu
             if isinstance(p, build.IncludeDirs):
                 result.append(p)
             elif isinstance(p, str):
+                if key == 'd_import_dirs' and os.path.normpath(p).startswith(self.environment.get_source_dir()):
+                    FeatureDeprecated.single_use('Building absolute path to source dir is not supported',
+                                                 '1.6.0', self.subproject,
```

Comment:
```suggestion
                                                 '0.45', self.subproject,
```
Did a little git archeology, this is an old deprecation!

Reply:
Done, many thanks for the archaeological research! This could've caused a lot of confusion for users had you not pointed this correction out.